Source/Extensions/Blazorise.SignaturePad/wwwroot/signaturepad.js

```
@@ -0,0 +1,153 @@
+            instance.sigpad.clear();
+            instance.sigpad.fromData(data);
+        }
+
+        if (options.velocityFilterWeight.changed) {
+            instance.sigpad.velocityFilterWeight = options.velocityFilterWeight.value;
+        }
+
+        if (options.dotSize.changed) {
+            instance.sigpad.dotSize = options.dotSize.value;
+        }
+
+        if (options.readOnly.changed) {
+            if (options.readOnly.value === true) {
+                instance.sigpad.off();
+            }
+            else {
+                instance.sigpad.on();
+            }
+        }
+
+        if (options.imageType.changed) {
+            instance.options.imageType = options.imageType.value;
+        }
+
+        if (options.imageQuality.changed) {
+            instance.options.imageQuality = options.imageQuality.value;
+        }
+
+        if (options.includeImageBackgroundColor.changed) {
+            instance.options.includeImageBackgroundColor = options.includeImageBackgroundColor.value;
+        }
+    }
+}
+
+export function setData(element, elementId, data) {
+    const instance = _instances[elementId];
+
+    if (instance && instance.sigpad) {
+        instance.sigpad.fromData(data);
+    }
+}
+
+export function clear(element, elementId) {
+    const instance = _instances[elementId];
+
+    if (instance && instance.sigpad) {
+        instance.sigpad.clear();
+    }
+}
+
+function registerToEvents(dotNetAdapter, instance) {
+    if (instance && instance.sigpad) {
+        instance.sigpad.addEventListener("beginStroke", (e) => {
+            dotNetAdapter.invokeMethodAsync("NotifyBeginStroke")
+        });
+
+        instance.sigpad.addEventListener("endStroke", (e) => {
+            const dataURL = getImageDataURL(instance.sigpad, instance.options);
+
+            dotNetAdapter.invokeMethodAsync("NotifyEndStroke", dataURL)
+                .catch((reason) => {
+                    console.error(reason);
+                });
```

Why is this swallowing an exception here??? Are we hiding a subtle bug? Also we should not console.error.

---

removed